client/client-multi/client-c/src/kaa/kaa_logging.c

```diff
@@ -104,24 +100,32 @@ static kaa_error_t remember_request(kaa_log_collector_t *self, uint16_t bucket_i
     return KAA_ERR_NONE;
 }
 
-
-
 static bool find_by_bucket_id(void *data, void *context)
 {
     KAA_RETURN_IF_NIL2(data, context, false);
-    return (((timeout_info_t *)data)->log_bucket_id == *((uint16_t *)context));
-}
+    timeout_info_t *timeout_to_find = context;
+    timeout_info_t *timeout_info = data;
 
+    if (timeout_info->log_bucket_id == timeout_to_find->log_bucket_id) {
+        *timeout_to_find = *timeout_info;
```

Comment:
To avoid context state changes, please use kaa_list_find_next() to get the necessary info and kaa_list_remove_at() to remove the element